server/odc-service/src/main/java/com/oceanbase/odc/service/notification/ChannelFactory.java

```diff
@@ -32,14 +29,13 @@
 @Service
 @SkipAuthorize("odc internal usage")
 public class ChannelFactory {
-    @Autowired
-    private UserRepository userRepository;
 
-    public Channel generate(ChannelConfig channelConfig) {
-        if (channelConfig.getType() == ChannelType.DingTalkGroupBot) {
-            return new DingTalkBotChannel(channelConfig, userRepository);
-        } else {
-            throw new NotImplementedException();
+    public MessageChannel generate(Channel channel) {
+        switch (channel.getType()) {
```

Comment:
verify `channel` and `channel#getType` are not null to avoid protential NPE

Reply:
I moved it into R4242NotificationPolicyMetaMigrate and notification-policy-metadata.yaml. This will only be executed once.